ext/standard/mail.c

```diff
@@ -199,6 +199,28 @@ PHPAPI zend_string *php_mail_build_headers(HashTable *headers)
 	return s.s;
 }
 
+/* {{{ php_mail_disable_flags
+   Remove all dangerous command line flags
+   from the sendmail shell command
+*/
+PHPAPI zend_string *php_mail_disable_flags(const char *str)
+{
+        const char* blacklist[4] = {"-O", "-C", "-X", "-be"};
+        size_t str_length = strlen(str);
+        char *return_str = emalloc(str_length + 1);
```

Comment:
Is this allocation really needed?

You can just allocate the zend_string directly, and iterate on the underlying char pointer.